src/installer/corehost/common/pal.unix.cpp

```diff
@@ -331,6 +332,55 @@ bool pal::get_temp_directory(pal::string_t& tmp_dir)
     return false;
 }
 
+bool pal::get_default_bundle_extraction_base_dir(pal::string_t& extraction_dir)
+{
+    if (!get_temp_directory(extraction_dir))
+    {
+        return false;
+    }
+
+    append_path(&extraction_dir, _X(".net"));
+    pal::string_t dotnetdir(extraction_dir);
+
+    // getuid() is the real user ID, and the call has no defined errors.
+    struct passwd* passwd = getpwuid(getuid());
+    if (passwd == nullptr || passwd->pw_name == nullptr)
+    {
+        return false;
+    }
+
+    append_path(&extraction_dir, passwd->pw_name);
+
+    if (is_read_write_able_directory(extraction_dir))
+    {
+        return true;
+    }
+
+    // Create $TMPDIR/.net accessible to everyone
+    if (::mkdir(dotnetdir.c_str(), S_IRWXU | S_IRWXG | S_IRWXO) == 0)
+    {
+        // In the above mkdir() system call, some permissions are strangely dropped!
+        // Linux drops S_IWO and Mac drops S_IWG | S_IWO.
+        // So these are again explicitly set by calling chmod()
+        if (chmod(dotnetdir.c_str(), S_IRWXU | S_IRWXG | S_IRWXO) != 0)
+        {
+            return false;
+        }
+    }
+    else if (errno != EEXIST)
+    {
+        return false;
+    }
```

Comment:
What happens if you run a program with `$TMPDIR/.net` already created, but with the permissions different than the one you set with `chmod()` because someone went there and changed them?

Reply:
That would be too bad. We just provide the `$TPMDIR/.net/$UID` as the default for `$DOTNET_BUNDLE_EXTRACTION_BASE_DIR`. If the default directory is not available/accessible on a system, then the user should set `DOTNET_BUNDLE_EXTRACTION_BASE_DIR` to an alternate location.